### DISCUSSION TITLE
Generating a 'dual' protein binder

### DISCUSSION THREAD

**Discussion by KodyKlupt** [RosettaCommons/RFdiffusion#268] [Q&A]
Looking for some advice on generating a binder that links protein 1 (chain A) and protein 2 (chain B), separated by a fixed distance in coordinate space. Perhaps can also consider this as a fusion protein (though not one sequence chain) where the middle protein is the de novo binder to both terminals.

Wondering what, if possible, the contigs would look like? Given a PDB file with chain A and B can RFDiffusion support e.g. A1-100/0 B1-100/0 50-100 (both proteins 100 residues, with a 50-100 a.a. binder)?

Thanks!

**Comment by roccomoretti** [comment]
RFdiffusion works on a "residue gas" basis, with chains more-or-less functioning purely on connectivity annotation basis. As such, it should be robust to having the known context being provided as multiple chains.

Your contig specification should work as you expect it to. I will note, however, that there's nothing in the contig specification which requires the generated protein to be built _between_ chains A & B. You probably want to provide hotspot residues on both chains to encourage RFdiffusion to generate backbones which contact both chains.

**Reply by ii44545** [reply]
In this case, A1-100/0 B1-100/0 50-100, the C-terminus of chain A will be connected to the N-terminus of chain B, so how to solve this problem?

**Reply by ii44545** [reply]
I guess that the code just recognize the last /0 in the contigmap.contigs, but I just dont know

**Reply by ii44545** [reply]
Sorry I dont mean this. For example peptide A, B and C combine together and I wanted to form a protein(50-100) to bind that complex, so I used A1-100/0 B1-100/0 C1-100/0 50-100, and the final result shows that there are 2 chains, one of them is the A1-100,B1-100,C1-100 complex named "chain A" and the newly formed chain is "chain B". When I put the pdb file in the chimera, I found that the C-terminus of A will be connected to the N-terminus of B and the C-terminus of B will be connected to the N-terminus of C in "chain A" but thats not how protein ABC actually connect to each other. So I really want to know if there is a better way to solve my personal question.

**Reply by rclune** [reply]
Can you provide the full input and output for what you have attempted? 

Does RFdiffusion change the structure of the given A, B, C peptides other than joining their termini?

**Reply by ii44545** [reply]
No the structure doesnt change, the model just link their teminus together in the final pdb file.
